### ISSUE TITLE
Cannot read props in async resource fetcher

### ISSUE THREAD

**Issue by elsassph** [solidjs-community/eslint-plugin-solid#195]
**Describe the bug**

Can't read props in an async resource fetcher function. I'd think it is expected that the function should be considered untracked.

**To Reproduce**

```typescript
const [res] = createResource(input, async (input) => { // <-- solid/reactivity for the function
   const thing = props.thing;
   const result = await fetchSomething(input, thing);
   return result;
});
```

**Expected behavior**

I shouldn't be forced to disable the rule for the function.

**Screenshots**

**Environment (please complete the following information):**

- OS: [e.g. Mac OS 11, Windows 10]
- Node version (`node --version`):
- `eslint-plugin-solid` version (`npm list eslint-plugin-solid`/`yarn why eslint-plugin-solid`): 0.14.5
- `eslint` version (`npm list eslint`/`yarn why eslint`): 8.57.1

**Additional context**

- [ ] I would be willing to contribute a PR to fix this issue

**Comment by ryansolid** [comment]
Confirmed on current main: the async fetcher of `createResource(source, fetcher)` is still treated as a tracked scope, so props reads inside it warn. The fetcher receives its source value as an argument and is not itself tracked — reads inside it should be allowed. Queued as a 0.16.1 false-positive fix together with #199.

**Comment by ryansolid** [comment]
Fixed in v0.16.1. In `createResource(source, fetcher)` the fetcher is now treated as an untracked called function — it may be async and read current values freely. The source remains a synchronous tracked scope.
